[max][layout] Add load_scalar method to LayoutTensor for scalar element access

KrxGu opened the pull request:
Fixes #4847

## Problem

Currently, `LayoutTensor.__getitem__` returns `Self.element_type` which is `SIMD[dtype, element_size]`. Since `element_size` isn't guaranteed to be 1 (e.g., in tiled/vectorized layouts), users working in generic contexts often get a SIMD vector when they just want a single scalar value. This leads to confusion and awkward workarounds like `tensor[i, j][0]`.

## Solution

This PR adds a `load_scalar` method that explicitly returns `Scalar[dtype]` for cases where you need a single value:

```mojo
# Before: have to extract lane 0 manually
var val = tensor[i, j][0]

# After: clear intent, returns Scalar directly  
var val = tensor.load_scalar(i, j)
```

## Changes

Added `_load_scalar_offset` private helper that reuses existing `_load_offset `and returns lane 0
Added `load_scalar[*Tys: Indexer]` for variadic indices like `tensor.load_scalar(i, j)`
Added load_scalar(RuntimeTuple) overload for coordinate-based access
Added tests covering static layouts, dynamic layouts, and RuntimeTuple coordinates

## Design note

I went with a named method rather than a `__getitem__` overload because having two methods with identical parameter signatures but different return types would cause ambiguity issues. The `load_scalar` name also makes the intent explicit at the call site, which seems cleaner for this use case.

KrxGu committed:
Add load_scalar method to LayoutTensor (#4847)

Add scalar element access for LayoutTensor that returns Scalar[dtype]
instead of SIMD[dtype, element_size], addressing user confusion in
generic contexts where element_size > 1.

- Add _load_scalar_offset private helper
- Add load_scalar[*Tys: Indexer] for variadic index access
- Add load_scalar(RuntimeTuple) for coordinate-based access
- Add comprehensive tests

martinvuyk commented on max/kernels/test/layout/test_layout_tensor_load_scalar.mojo:
```diff
@@ -0,0 +1,211 @@
+    print("== test_load_scalar_static_layout")
+
+    comptime layout = Layout.row_major(2, 3)
+    var storage = UnsafePointer[Float32].alloc(6)
```
You can just use a `List` or an `InlineArray` as the backing storage, no need to use a pointer

martinvuyk commented on max/kernels/test/layout/test_layout_tensor_load_scalar.mojo:
```diff
@@ -0,0 +1,211 @@
+    test_load_scalar_with_runtime_tuple()
+    test_load_scalar_matches_getitem_lane0()
+    test_load_scalar_vectorized_element_size_gt_1()
+    print("All tests passed!")
```
please remove the print statements and just use `from testing import TestSuite` and auto import the tests, you can see other places in the stdlib where it's used

martinvuyk reviewed (commented):
Hi, thanks for contributing!

I'll ask a question I have regarding the actual need for this in the linked issue, but I'll leave a couple of comments of things that stand out to me to give you some feedback

KrxGu committed:
refactor: use InlineArray and TestSuite in load_scalar tests

Address review feedback:
- Replace UnsafePointer.alloc() with InlineArray for safer memory management
- Use TestSuite.discover_tests pattern instead of manual print statements
- Remove manual test progress prints (PASSED messages)
